### PR TITLE
feat: add badge icon for work profile apps

### PR DESCRIPTION
**Pull request by kaanelloed**
Add the work profile badge.

workprofilebadge

### CODE REVIEW

**Comment by falzonv** [comment]
Hello,

Interesting, I didn't know about getUserBadgedIcon.
However, since we are discussing about icon packs in other tickets, this would conflict with them by using the actual app icon instead of the replacement from the icon pack.

Currently there is indeed no immediate way to distinguish regular apps and work profile apps, but users can manually apply some workarounds such as putting all work profile apps in a dedicated folder, or renaming work profile apps to add something at the end of their names (or make them start with a "z" so they are at the end of the list).

It might be possible to handle it better (ex: adding a badge ourselves) but this would add much more complexity than this request and is not something I want to manage to be honest (see project status in ticket #314).

Sorry this will not be implemented.

Best regards.

**Comment by kaanelloed** [comment]
> Interesting, I didn't know about getUserBadgedIcon. However, since we are discussing about icon packs in other tickets, this would conflict with them by using the actual app icon instead of the replacement from the icon pack.

@falzonv The method I've chosen adds the badge to the icon provided, and doesn't use the default application icon.

File: app/src/main/java/com/vincent_falzon/discreetlauncher/core/ApplicationsList.java [hunk 1]
```diff
@@ -27,6 +27,7 @@
 import android.content.SharedPreferences ;
 import android.content.pm.LauncherActivityInfo ;
 import android.content.pm.LauncherApps ;
+import android.content.pm.PackageManager;
```
Comment (falzonv) [inline app/src/main/java/com/vincent_falzon/discreetlauncher/core/ApplicationsList.java:30]:
```suggestion
import android.content.pm.PackageManager ;
```

File: app/src/main/java/com/vincent_falzon/discreetlauncher/core/ApplicationsList.java [hunk 2]
```diff
@@ -110,6 +111,8 @@ public void update(Context context)
 		UserManager userManager = (UserManager)context.getSystemService(Context.USER_SERVICE) ;
 		List<UserHandle> userProfiles = userManager.getUserProfiles() ;
```
Comment (falzonv) [inline app/src/main/java/com/vincent_falzon/discreetlauncher/core/ApplicationsList.java:113]:
```suggestion
```

File: app/src/main/java/com/vincent_falzon/discreetlauncher/core/ApplicationsList.java [hunk 3]
```diff
@@ -110,6 +111,8 @@ public void update(Context context)
 		UserManager userManager = (UserManager)context.getSystemService(Context.USER_SERVICE) ;
 		List<UserHandle> userProfiles = userManager.getUserProfiles() ;
 
+		PackageManager pm = context.getPackageManager() ;
```
Comment (falzonv) [inline app/src/main/java/com/vincent_falzon/discreetlauncher/core/ApplicationsList.java:114]:
```suggestion
		PackageManager packageManager = context.getPackageManager() ;
```

File: app/src/main/java/com/vincent_falzon/discreetlauncher/core/ApplicationsList.java [hunk 4]
```diff
@@ -136,6 +139,10 @@ public void update(Context context)
 						if(color_tint == 0) icon = activity.getIcon(0) ;
 							else icon = applyColorTint(resources, activity.getIcon(0), color_tint) ;
 					}
+
+				if (userHandle != null)
+					icon = pm.getUserBadgedIcon(icon, profile) ;
```
Comment (falzonv) [inline app/src/main/java/com/vincent_falzon/discreetlauncher/core/ApplicationsList.java:144]:
```suggestion
					icon = packageManager.getUserBadgedIcon(icon, profile) ;

```

File: app/src/main/java/com/vincent_falzon/discreetlauncher/core/ApplicationsList.java [hunk 5]
```diff
@@ -136,6 +138,11 @@ public void update(Context context)
 						if(color_tint == 0) icon = activity.getIcon(0) ;
 							else icon = applyColorTint(resources, activity.getIcon(0), color_tint) ;
 					}
+
+				if (userHandle != null)
+					icon = packageManager.getUserBadgedIcon(icon, profile) ;
+
+
```
Comment (falzonv) [inline app/src/main/java/com/vincent_falzon/discreetlauncher/core/ApplicationsList.java:145]:
```suggestion
				// Resize the icon to the user-defined size
```

File: app/src/main/java/com/vincent_falzon/discreetlauncher/core/ApplicationsList.java [hunk 6]
```diff
@@ -136,6 +138,11 @@ public void update(Context context)
 						if(color_tint == 0) icon = activity.getIcon(0) ;
 							else icon = applyColorTint(resources, activity.getIcon(0), color_tint) ;
 					}
+
```
Comment (falzonv) [inline app/src/main/java/com/vincent_falzon/discreetlauncher/core/ApplicationsList.java:141]:
```suggestion

				// Add a badge to the chosen icon if the app is in a work profile
```

File: app/src/main/java/com/vincent_falzon/discreetlauncher/core/ApplicationsList.java [hunk 7]
```diff
@@ -136,6 +138,11 @@ public void update(Context context)
 						if(color_tint == 0) icon = activity.getIcon(0) ;
 							else icon = applyColorTint(resources, activity.getIcon(0), color_tint) ;
 					}
+
+				if (userHandle != null)
+					icon = packageManager.getUserBadgedIcon(icon, profile) ;
+
```
Comment (falzonv) [inline app/src/main/java/com/vincent_falzon/discreetlauncher/core/ApplicationsList.java:144]:
```suggestion
```

**Comment by falzonv** [comment]
> **falzonv:** Ideally the badge should also be affected by the possibly user-defined color tint if no icon pack is used, I will have a look on what needs to be changed for that.

This in fact a bit more convoluted than I thought for a feature most people probably never use (the color tint), let's forget about this point and keep it simple.

File: app/src/main/java/com/vincent_falzon/discreetlauncher/core/ApplicationsList.java [hunk 8]
```diff
@@ -109,6 +110,7 @@ public void update(Context context)
 		// Retrieve the list of user profiles
 		UserManager userManager = (UserManager)context.getSystemService(Context.USER_SERVICE) ;
 		List<UserHandle> userProfiles = userManager.getUserProfiles() ;
+		PackageManager packageManager = context.getPackageManager() ;
```
Comment (falzonv) [inline app/src/main/java/com/vincent_falzon/discreetlauncher/core/ApplicationsList.java:113]:
```suggestion
		PackageManager apkManager = context.getPackageManager() ;
```

Just noticed that for some reasons I don't remember, I called the package manager "apkManager" in several other places of the launcher, let's do the same here for consistency.

File: app/src/main/java/com/vincent_falzon/discreetlauncher/core/ApplicationsList.java [hunk 9]
```diff
@@ -136,6 +138,11 @@ public void update(Context context)
 						if(color_tint == 0) icon = activity.getIcon(0) ;
 							else icon = applyColorTint(resources, activity.getIcon(0), color_tint) ;
 					}
+
+				// Add a badge to the chosen icon if the app is in a work profile
+				if(userHandle != null) icon = packageManager.getUserBadgedIcon(icon, profile) ;
```
Comment (falzonv) [inline app/src/main/java/com/vincent_falzon/discreetlauncher/core/ApplicationsList.java:143]:
```suggestion
				if(userHandle != null) icon = apkManager.getUserBadgedIcon(icon, profile) ;
```
Just noticed that for some reasons I don't remember, I called the package manager "apkManager" in several other places of the launcher, let's do the same here for consistency.

**Pull request merged**